### ISSUE TITLE
Intro to audit

### ISSUE THREAD

**Issue by lauracowen** [OpenLiberty/docs#213]
https://github.com/OpenLiberty/docs/blob/audit/concept/audit-runtime-apps.adoc

**Comment by lauracowen** [comment]
Contact:
Emily Tuckowski

Background:
The Audit feature provides a load of events (data emitted from Open Liberty) in production so that Ops teams can gather the data for audit purposes (ensuring integrity of systems, data, etc).

**Comment by lauracowen** [comment]
Please get Emily to review and approve this topic. Check it reads okay too.

**Comment by Rwalls1** [comment]
@emilytee Please review: https://github.com/OpenLiberty/docs/blob/audit/concept/audit-runtime-apps.adoc

**Comment by emilytee** [comment]
@Rwalls1 , looks fine.   I don't see a review record for me tho.

**Comment by lauracowen** [comment]
Can you build this topic to the drafts site pls?

**Comment by Rwalls1** [comment]
@lauracowen Please review, the topic is now on the draft site: https://draft-openlibertyio.mybluemix.net/docs/ref/general/#/docs/concept/audit-runtime-apps.html

**Comment by lauracowen** [comment]
- The first sentence reads oddly - can you rephrase it?
- The json examples are very long. Can you see if you can remove the white space between the lines at least? then provide more narrative about the flow of the steps between each example snippet.
- The article ends abruptly (this is partly because it was lifted as a section from a longer PDF) - can you read it through and think about how the article as a whole would make sense to the reader (the ending but also the earlier bits). eg the introduction to the code example has a random sub-sub-heading in it that seems unnecessary. Can you read it through with an end-user hat on and assess how useful it is and what info it would need to be more useful/clear? The purpose of this topic is to introduce the concept of auditing. eg what does it mean to be available in cloud and in dashboards? can you give a concrete example of an audit trail scenario that you might use if someone asked you verbally to explain the point of auditing? (ask Emily for help - I don't expect you to pluck it out your head :) )

**Comment by lauracowen** [comment]
https://draft-openlibertyio.mybluemix.net/docs/ref/general/#/docs/concept/audit-runtime-apps.html

Can you ask Emily to check the draft and then add a comment to this issue if she's fine with it being published, please?

**Comment by chirp1** [comment]
Hi Richard, it looks like this topic is one of the topics that was overwritten on the draft server. Would you put it back in the draft branch? Then, let me know when it is displaying so that I can review it? - Thanks!

**Comment by chirp1** [comment]
Hi Richard,
I'm reviewing what you have on your branch since the content isn't on the draft server.

- Please rewrite the opening paragraph as it reads awkwardly. Use the our workshops on short descriptions and the IBM _Developing quality technical information_ on-line book as a guide. In the first sentence change  "The Audit (audit-1.0) feature" to "theaudit-1.0 feature". Get rid of extra words in the first sentence. Make the different aspects of the second sentence parallel. 

- Use complete sentences except for the first clause in the tables. For instance, "Any event that occurs within the environment..." is not a complete sentence.

- Do not have any quotes except those in development code because quotes are accessibility violations.

- Rewrite this sentence: "The data in an event can record the WHO, WHAT, WHEN, ON WHAT, WHERE, and optionally, FROM WHERE and WHERE TO.". Rewrite it in the style of "The data is an even can record, who initiated the action, 

- Run Acrolinx on the topics and fix any outstanding issues. I see a few things that Acrolinx is flagging. 

- Replace ".etc" with "and so on".

- Don't use parentheses For instance, change The user ID (user1)" to "The user1 user ID".

- are onwhat, fromwhere, towhere in the W column really components? If not, change them to "on what" "from where", and "to where". In the description, don't use the component names, use the plain english wording.

**Comment by lauracowen** [comment]
@chirp1 I can't find the file to check exactly but from your comments above, for the feature name, can you use the macro (@Charlotte-Holt and @dmuelle know how) so that it consistently resolves the feature name in the build and it generates a link to the feature topic?

**Comment by lauracowen** [comment]
(moved back to In Progress while Karen's feedback is addressed)

**Comment by Rwalls1** [comment]
@lauracowen Please review: https://draft-openlibertyio.mybluemix.net/docs/ref/general/#/docs/concept/audit-runtime-apps.html

**Comment by Rwalls1** [comment]
@emilytee has reviewed and approved the changes to the topic: https://draft-openlibertyio.mybluemix.net/docs/ref/general/#/docs/concept/audit-runtime-apps.html

**Comment by chirp1** [comment]
@Rwalls Hi Richard, would you check with Emily about the component names in the table? Her .adoc file at the top of this issue has variations of the component names. What you have in the table must match what is in the code. For instance, is it FromWhere, fromWhere, or something else? Also, would you spruce up the phrasing in the description column? See my previous comments about wording- or ask me and we can discuss.

**Comment by lauracowen** [comment]
@Rwalls1 Hi Richard, just so you know, I've been reviewing all three of your Audit topics this week (got held up by release problems elsewhere) and will post feedback in the issues today.

**Comment by lauracowen** [comment]
Thank you. I've finally reviewed this topic (sorry for the long delay) and provided some feedback on the content, in addition to the changes Karen requested above.

The aim of this topic is to introduce the concept of the auditing applications running on Open Liberty. The target audience is an operations person who needs to audit what is happening in their production systems (maybe for legal requirements or for security reasons).

**Intro paragraph**
- [ ] This intro paragraph could do with being a bit more informative, such as defining what is meant by 'audit'. So, it needs to briefly say why the application (or runtime? or what?) needs to be audited. ie state whatever the problem is that the Audit feature is solving (why does anyone care about these audit events?).  Give a concrete example even (eg something to do with security as the example later on is about logging in). Then talk about how the Audit feature addresses this problem by, when enabled, capturing audit events emitted by the environment in the audit log.
- [ ] I think it then needs to briefly introduce the idea that things (what things? server runtime? application? operating system?) emit events when something (what?) happens while an application is running on an Open Liberty server. Then that the Audit feature can capture these events.
- [ ] Remove the "Auditing the server runtime environment and applications" heading completely and just make this part of the introductory section above (might need rewording to blend it all together).
- [ ] Although Karen said "In the first sentence change "The Audit (audit-1.0) feature" to "theaudit-1.0 feature". ", can you switch back to saying "Audit (audit-1.0) feature" the first time you say it (then "the Audit feature" after that). The reason is that Alasdair has been trying to get people to refer to the features by name in blog posts etc as they're often more readable than the technical name for the feature. Also, the second sentence refers to the feature by name ("Audit feature") so it makes sense to use that term in the first sentence too to link the two sentences together.

**Audit key terms section**
- [ ] Just remove this section completely (the heading and the sentence beneath it). I don't think it provides any value to the topic. Then the topic can just go straight into describing the main concepts that the operations person needs to understand about auditing.


** Audit event section**
- [ ] I think change the heading to "Audit events". Otherwise, I think the text in this section works okay.
- [ ] I agree with Karen about the descriptions in the first table, though I think having the key words highlighted in quotes doesn't help with comprehension. I don't think there's any technical reason why you shouldn't just provide plain English wording for each of the descriptions like you would in any other table. So, for example, the `What`row's description would be just "What activity occured and the result of the activity."; the `onWhat` row's description could be "The resource targetted by the activity."; the `ToWhere` row's description could be "The target of the action." I think a plain English approach is much more readable (especially for people where English isn't their first language) and understandable in this case and I don't think it helps to try to use the component name in the description.


**Audit outcome section**
- [ ] I think pluralise this heading too: "Audit outcomes".
- [ ] I'm not sure this section makes much sense. I think there needs to be a better explanation of what an "outcome" is? Is it something that happens as a result of the event? Is it information that is in the event that describes the consequence of whatever happened that triggered the event in the first place? Why does the user need to understand the outcomes of events - what do they do with that information? Then maybe an example of why you might get one of those outcomes.
- [ ] The third sentence ("Most audit events…") doesn't make much sense - as reader, I'm left wondering "so what?". If this is important information for the reader to know, it needs to be clear why they're being told - and, in particular, why if that's the normal outcome types, the Audit feature here provides more - what's the point in having extra outcomes?

**Audit handler**
Right, I think this is where you might need to discuss it further with the SME.  From reading this section and the generated Audit feature doc, it's not very clear what the user is meant to understand from the term "default file handler".
- [ ] This sentence "The audit handler is an implementation of the collection manager handler interface that outputs the audit event to a repository" is quite difficult to understand. I don't think it's the wording but the terms and concepts being used. Can you go back to Emily and ask her about it. It's not clear whether the user actually needs to care about the "collection manager handler interface" - as it's not been mentioned anywhere else, I suspect not. If the majority of users just need to configure the supplied audit handler, this section should just talk in those terms. eg "The audit handler outputs the audit event to a repository, such as a flat file, a database, or the cloud."
- [ ] Based on the text in the Examples section of https://draft-openlibertyio.mybluemix.net/docs/ref/feature/#audit-1.0.html, it sounds like most users just configure a (or maybe multiple, I'm not quite sure?) of the supplied audit file handler (ie an `auditFileHandler` section in the server.xml). But check that detail. If I'm right, though, just talk about how the user (ops person) has to configure, say, one or more instances of the auditFileHandler in the server.xml (link to the Audit feature generated topic - ask @Charlotte-Holt for how to do that with a macro instead of hard-coding the link). If I'm right, don't talk about implementations of the collection manager handler interface unless Emily says that all users of the Audit feature need to write some Java to implement that interface in order to do anything with audit (I suspect that's not the case).
- [ ] I think then change the "Audit handler" heading to "Examples" or similar.
- [ ] Then check that the explanations of the two examples makes sense. Start the Examples section with something like "The following examples show the audit events emitted when a user successfully logs in to a website and when their login fails."
- [ ] In the "Successful form login…" sub-section, give a bit more of a summary of what the example does - the examples are so long (in terms of the space they take on the page) that it's hard to get an overview of them at a glance, so providing a short summary of the steps at the top of each example would be useful.

Please ask me if anything doesn't make sense or if you disagree with any of the comments above, or talk to Charlotte.

In terms of SME, for technical questions, ask EmilyT - but be specific in the questions you ask her rather than just asking her to review the whole topic. Make suggestions to her for better wording etc if necessary.

I think it would also be useful for you to ask Don Bourne ( @Charlotte-Holt has worked with him recently) to review this topic if he has time. Ask him if he thinks there's anything else that should be included for understanding how to use the Audit capability in Liberty, or if there's anything you should approach differently etc, as well as any specific questions you might have. His team is responsible for the operations/monitoring types of capability in Liberty and he makes really useful suggestions on docs if he has time.

**Comment by Rwalls1** [comment]
@lauracowen Please review the latest update: https://draft-openlibertyio.mybluemix.net/docs/ref/general/#audit-runtime-apps.html. For the "Successful form login" section,  when I spoke with Emily, I inferred that the current explanation before each example does explain what's happening as each example demonstrates a different step. I can with work with her to try to provide a more thorough explanation of each step before each example but I think the overall structure for that section is fine.

**Comment by Rwalls1** [comment]
@donbourne Please review: https://draft-openlibertyio.mybluemix.net/docs/ref/general/#audit-runtime-apps.html

**Comment by lauracowen** [comment]
- there are `[.underline]` characters in some of the code snippets still - can you do a search through the whole file? I think these were in the original .adoc file I gave you because they must've copied oddly from Emily's original PDF - sorry about that.
- Can you address this comment: 
  The third sentence ("Most audit events…") doesn't make much sense - as reader, I'm left wondering "so what?". If this is important information for the reader to know, it needs to be clear why they're being told - and, in particular, why if that's the normal outcome types, the Audit feature here provides more - what's the point in having extra outcomes?
  Why do most audit events only emit those outcome types? Why does the reader need to know this?
- typo: "is provided is provided"
- the Audit handler section looks simpler now, thanks. Can you say something about how they need to configure the audit handler in the server.xml and link to the Audit feature gendoc?
- Then, if you don't have a single "Examples" heading (which makes sense the way you have the Audit handler intro now), can you maybe suffix each of the example scenario headings with "example" to make it clear that all the sub-headings beneath "Audit handler" are examples?

Otherwise, looks good, thanks for sticking with this. If you can address the comments above we can finish this off.

**Comment by Rwalls1** [comment]
@lauracowen Please review the latest updates: https://draft-openlibertyio.mybluemix.net/docs/ref/general/#audit-runtime-apps.html

**Comment by lauracowen** [comment]
- "in an on-premises audit log." does that mean it doesn't work for cloud deployments?
- following on from our discussion about the other event topics with Don, can you ask Emily (or make it up yourself) and anonymised and shorter example URL (ie so it doesn't use a real IBM server, and so it doesn't use our real build path).
- The `#` that appear multiple times in the paths (of the `id`, `target`, and `name` fields) look odd - are they correct?
- can you highlight file names with backticks eg `server.xml`, `audit.log`?

Otherwise, I think it's ok for publishing. :)

**Comment by donbourne** [comment]
apologies for delayed reply on this -- I think this should be reviewed by @emilytee , since she owns audit, not me (I'm involved in JSON logging of which Audit is a part, but since this topic is purely on audit it's something Emily owns).

**Comment by Rwalls1** [comment]
@lauracowen Please review the latest update: https://draft-openlibertyio.mybluemix.net/docs/ref/general/#audit-runtime-apps.html

**Comment by lauracowen** [comment]
Thank you - well done on getting this finished. :)

**Comment by Rwalls1** [comment]
@NottyCode Please review: https://draft-openlibertyio.mybluemix.net/docs/ref/general/#audit-runtime-apps.html

**Comment by NottyCode** [comment]
I don't like the introduction to this topic. It feels like it is overly long, and doesn't really give a good overview of what audit is and why you would care about it. It talks about environment without defining what it is and it makes it sound like it is a way to identify and solve configuration errors (which it isn't). I've put some feedback below about specific items, but think it needs to be revised significantly.

1. I don't understand what this sentence is trying to say:

> By enabling audit support with the Audit feature, you can assess the configured environment with the Audit data that is provided in an audit log.

The bit up to the first comma seems somewhat redundant, but it is the bit after the comma that I don't understand.

2. I think this sentence is overly complicated way of describing what auditing is for and given the following sentence which seems much simpler and easier to follow I think it is redundant.

> Auditing helps you to identify and understand the behavior of users and applications by using audit logs that consist of data that reveals different interactions with the server runtime environment.

3. I wasn't aware that audit logs could discover configuration errors. In fact this was a total surprise to me because I'm often asked by people how do troubleshoot configuration issue and I've never once suggested using audit for this.

> While the functions of audit logs can be relegated to only troubleshooting errors within the configuration of the environment, the Liberty audit feature can be used to extend those capabilities to enable the observation of the environment to help prevent any potential security risks.

4. The article talks a lot about configured environments, but doesn't define what they are. Given all audit does is report on things happening in Liberty and a configured environment by its nature is larger I think this is over promising vs what we do.

----

This might be personal preference, but I think the introduction should introduce the key concepts so when you get to Audit Events you don't have a title that is devoid of context.

5. I think the following sentence is quite long and as a result I found it difficult to really follow or understand. Perhaps a bulleted list might be easier to follow:

>An audit event is any event that occurs within the environment that can be represented by who attempted to perform the action, when they tried to perform it, what resource did they try to perform the action on, and the outcome.

6. The second paragraph seems to repeat information in the first. It feels repetitive and means my eyes just skim over the entire sentence.

> The data in an event can record who initiated the action, what initiated the action, when the action was initiated, on what the action was initiated, where the action was initiated, and optionally from where did the action initiate and to where did the action target.

7. The "W" component header in the table is impossible to read on my screen. It might be a font choice. Not sure why the W is in quotes. Is the 7 W's an industry term? It isn't one I've come across before.

8. Same comment for audit outcome and audit handler as for audit events. I think the intro should define these and the relationship and then the more sections should cover them in more depth.

9. I wouldn't use must here, it makes it sound like we are making a demand of the user. I would have just said "you configure the auditFileHandler element". I also wouldn't use the < or > around it.

10. The examples have \ markers in the JSON and there shouldn't be any.

11. The examples are not syntax highlighted. I see the source attempts this, but I'm guessing source,json isn't right.

12. What are the examples useful for? I just see a ton of json structure with very little description. We don't define the structure for the events we just have a bunch of examples without explanation of what each entry is. 

> To configure the audit file handler, you must configure the <auditFileHandler> element in the server.xml file.

**Comment by Rwalls1** [comment]
@lauracowen Please review, what are your thoughts on Alasdair's comments?

**Comment by Rwalls1** [comment]
@lauracowen Please review, I think I've addressed most of the feedback, what are your thoughts on the 7 W's table and the usefulness of the examples? I wasn't quite sure how to address those two points without further discussion.

**Comment by lauracowen** [comment]
@Rwalls1 

Regarding the 7 Ws, I think you need to ask Emily if it's an industry thing. If it is, provide a link to wherever it's described online. If it isn't, just don't say "are also referred to as...etc". In the table column header, just say "Component" (though I'm not sure "component" is a great term for it but if that's what they use, fine).

Regarding @NottyCode's comment about the usefulness of the examples. I think each event example is so long, you can't really see what's important/relevant to observer in them. The first and second examples are quite similar for most lines but there are some lines, like the reason code line, that are different. I think it would be maybe more useful if the examples were cropped down to just the informative lines, with a brief explanation of what the lines indicate. @NottyCode would you agree or do you not think they provide any useful help at all?

For fixing the syntax highlighting in the examples, if `source,json` doesn't do anything (eg just shows as blue on blue like it is now) try using `source,javascript` and see if that works any better.

The examples need the file paths shortening and anonymising. eg `/Users/sample/libertyGit/WS-CD-Open/dev/build.image/wlp/usr/` is a real path that we use internally and it's unnecessarily long for an example. Something like `/Users/sample/wlp/usr/`would be a lot more concise and should be fine. But make sure you update them all consistently.

Also, Emily needs to update the examples. They currently contain this statement ` "agent": "Apache-HttpClient/4.1.2 (java 1.5)"` which says that the app was running on Java 5. Liberty (let alone Open Liberty) never supported Java 5 (or 1.5). She needs to make sure the examples are relevant.

Minor thing: Can you remove the angular brackets from around element names in the text (eg `auditFileHandler` not `<auditFileHandler>`).

**Comment by Rwalls1** [comment]
@lauracowen Please review, I think I've addressed most of the feedback, I spoke with Emily about condensing the examples and she prefers to leave them as is due to the potential to mislead users about what the actual event record looks like, so I didn't condense for now as we may need to discuss further: https://draft-openlibertyio.mybluemix.net/docs/ref/general/#audit-runtime-apps.html

**Comment by dmuelle** [comment]
Hi Richard- this topic came up on our call with Laura this morning while looking at observability stuff in general. I wonder if instead of condensing the individual code examples, we could remove the first two examples under `Failed form login authentication example`, since they are essentially identical to the first two examples under `Successful form login flow to access a servlet application example`, ie both show the redirect and login form and so could apply to both successful and failed outcome examples. Would require some minor editing of the example descriptions but would also make the topic more concise.

**Comment by Rwalls1** [comment]
@emilytee What are your thoughts on the proposed update above?

**Comment by emilytee** [comment]
@Rwalls1 emilytee is convinced this doc will never go out!   I'm fine with @dmuelle's suggestion.   The original documentation was written with serviceability in mind, and reducing the number of questions/PMRs from our customers.    Hopefully the suggestion to remove the samples doesn't cause and questions/PMRs to get generated.

**Comment by Rwalls1** [comment]
@lauracowen Please review the latest update: https://draft-openlibertyio.mybluemix.net/docs/ref/general/#audit-runtime-apps.html

**Comment by lauracowen** [comment]
ok, thanks Richard. I'll sign this off now.

**Comment by ManasiGandhi** [comment]
#peer review

@rw2513 The topic looks good. A few suggesions,

- Acrolinx is flagging the word 'performed.'
- In the first sentence under 'Audit handler', can you change 'back-end repository' to 'backend repository'? In the same paragraph, two sentences begin with the same words, 'by default',  "By default, the audit file handler writes audit records to an audit log file until the file reaches 20 MB size. The audit log file is then archived and a new audit log file is started. By default, the maximum number of archived audit log files is 100 after which the oldest archived log file is overwritten." Can you reword the beginning of one of the sentences. Maybe begin the second sentence with 'Also, ...' and then mention 'by default' somewhere else in the sentence.
- In the second paragraph under 'Successful form login flow to access a servlet application example' it says, "The “session” field in the audit event records the HTTP session for this user interaction:" Are the quotes on “session” required? If not, can you take them off, as they may cause an accessibility error.
- Under 'Failed form login authentication example', same  point about "session"quotes.

**Comment by Rwalls1** [comment]
@ManasiGandhi Thanks, I've addressed your feedback.

**Comment by Rwalls1** [comment]
New link to topic based on Laura's guidance: https://draft-openlibertyio.mybluemix.net/docs/ref/general/#audit-logs

**Comment by chirp1** [comment]
Hi Richard,

Nice job on the topic!

Here are my comments:
- The title and intro don't match. Additionally, the intro paragraph isn't about the topic. The title of "Audit logs" doesn't seem to fit the contents of the topic. The topic seems to be an overview of audit. Redo the title and the intro paragraph. Refer the IBM style guide and the info that we used in the workshops on how to write a short desription.
- I see multiple inconsistencies in what terms are called. For example, "user password" and "password" are names of a term that mean the same thing. So, when a writer is developing a topic, only one of the two names would be used in a topic and across topics. So, go through your topic and make sure that all your terms are consistently named. 
- "A runtime environment provides a ...." has many thought in one sentence. Rewrite the information so that it is easier to understand. Break it into more than one sentence if necessary.
- Change the "The data that is generated ...." sentence to active voice. If you have other passive voice, evaluate and change to active voice as appropriate. 
- I see a variety of unnecessary words. Go through your topic and remove unnecessary words. If needed, rewrite sentence to get rid of them.
- Merge the second and third paragraphs into one paragraph as the information discusses what the topic is about. 
- Since the topic is a bit long, you could provide in-body links to the sections in the topic- either provide a links directly in the sentences in the current second and third paragraphs, or have a bulleted list of the section towards the top of the topic, with a link from the bulleted item to the appropriate section.
- In the Audit event section, the bulleted list of "components' are not components. The components are in the first column of the table. You could remove the bulleted list as it is redundant with the table.
- For the table, you need a title for it. David is investigating an consistent way for all you to write  asciidoc titles, captions, and alt text. Use the guidance if he has posted it. Otherwise, contact David.
- Also, for the table, describe what it is about. Also, describe what is in each column. Do these updates with the intent of providing useful information to anyone, and in particular to someone that is using a screen reader. You can start with something like "The following table....".
- For "... this happen", add a noun after "this".
- For the description column, determine whether the description should start with a caopital and change accordingly. 
- Also for the desrciption, the descriptions from row to row are not parallel. They should be. Some are stated in the form of a question others are not. Some are a mix of question and statement in the same sentence. 
- The "A repository can be an on-premises flat file, a database, or the cloud." sentence doesn't seem quite right grammatically and stylistically. Where is the database? On premise or elsewhere?  Do you mean "the cloud"? To figure out if the "the cloud" reference is correct, you might as an exercise remove the file and database references and then see if the "the cloud" reference makes sense.
- Change "reaches 20 MB size" to "reaches 20 MB".
- Change "confgi attributes" to "configuration attributes". (Spell out configuration whenever you use it.)
- The two examples in the Audit handler section are long. Let users know that two examples are available: Between the Audit handler paragraph and the first example, add a paragraph that tells the user that two examples are to follow. Provide links to each example.

**Comment by Rwalls1** [comment]
New link to topic: https://draft-openlibertyio.mybluemix.net/docs/20.0.0.9/audit-logs.html

**Comment by Rwalls1** [comment]
@chirp1 Can you provide an example of the unnecessary words you are referring to?

**Comment by chirp1** [comment]
@rwalls1 Hi Richard, Just to reiterate, in addition to putting questions in an issue for somebody, also slack the person as people don't necessarily see all the issue comments since they might be looking at many issues.  If you are still interested in discussing unnecessary words, would you slack me? Thanks!

**Comment by Rwalls1** [comment]
New link to topic: https://draft-openlibertyio.mybluemix.net/docs/20.0.0.10/audit-logs.html

**Comment by Rwalls1** [comment]
@chirp1 I've made updates, please review: https://draft-openlibertyio.mybluemix.net/docs/20.0.0.10/audit-logs.html

**Comment by Rwalls1** [comment]
PR: https://github.com/OpenLiberty/docs/pull/1690/files

**Comment by dmuelle** [comment]
Doc is now on vNext branch, will publish with 20.0.0.12 on 11.20

**Comment by chirp1** [comment]
Hi Richard, I have a few more comments:

- Change "Audit logs record events that help to analyze the server ..." to "Audit logs record events that help you analyze the server ...".
- For the Audit event components table, since you have more than one field in a cell, would you put each field on a separate line? 
- For the Description column in the table, I see both "activity" and "action". From the sentence starting wtih "An audit event is any action that occurs ..." I think that "activity" should be changed to "action" in the Description column.
- For each description would you add a question mark at the end?
- Change "When did this happen" to "When did the action happen".
- Change "Where did the activity get observed" to "Where was the activity observed".
- Change "Where did the action initiate" to "From where did the action initiate".
- For "Where was the target of the action", maybe "To where was the action targeted".
- Change the "Audit handler" title to "Audit file handler".
- Change "An audit file handler is provided by default to emit audit data..." to "An audit file handler emits audit data...".
- Remove the sentence that starts with "When the maximum number of archived audit logs.." because it is a bit awkward and it is unneeded. The next three sentences say the same thing and do so more clearly.
- Change "Audit logs can help provide important details ..." to "Audit logs can provide important details ...".
- Rewrite "The following examples show a successful form login flow to access a servlet application and a failed form login authentication." to create a bulleted list of the two links. I think the bulleted to list will provide a clearer structure to the user.
- For your description of the the "Successful form login flow ..." example, give the values for the security authentication, web servlet application, ID, etc.
- For Employee or Manager, either highlight them, since you capitalized them like they are in the example code, or make the words lower case and don't highlight them.
- For the "Failed form login authentication example", also add the values that are in the example code to the wording that you provided for the example description.

**Comment by Rwalls1** [comment]
@chirp1 Thanks, I've addressed your feedback: https://draft-openlibertyio.mybluemix.net/docs/21.0.0.8/audit-logs.html
In regards to your comments related to the "Successful form login" and "Failed form login" sections, I think the important values, such as `user1` user ID, `FormLoginServlet` application, and `baduser` user ID  are already referenced. Can you explain what other specific values that you think need to be mentioned in the sections?

**Comment by chirp1** [comment]
Hi Richard, 
- I'm OK with you your response to my comment about including important values for "Successful form login" and "Failed form login".
- I see a link to Audit 1.0. (1) It links to the draft server. The topic is not found when I click on the link for it. Would you fix the link? (2) I believe the style guidance for feature references is to not include the version of a feature as new versions can become available. So, maybe "Audit 1.0" > "the Audit feature", but check the OL style guidelines and follow them.
- For "What action occurred? and what was the result of the activity?", spruce the info to something like "What action occurred? What was the result of the action?".

**Comment by Rwalls1** [comment]
@chirp1 Thanks, I've addressed your latest feedback: https://docs-draft-openlibertyio.mybluemix.net/docs/21.0.0.8/audit-logs.html

**Comment by chirp1** [comment]
@Rwalls1 Hi Richard, The updates look good! Moving to Ready to Publish.

**Comment by Rwalls1** [comment]
Staging: https://staging-openlibertyio.mybluemix.net/docs/21.0.0.8/audit-logs.html

**Comment by Rwalls1** [comment]
Updates have been merged to Vnext and will publish with 21.0.0.8
